### DISCUSSION TITLE
Can't login to portal after clean docker installation

### DISCUSSION THREAD

**Discussion by mosespray** [vinyldns/vinyldns#1266] [General]
Hello,

I just followed the quickstart instructions to install vinyldns on ubuntu 22.04. 
Everything appeared to go correctly,   However, when I try to connect to the portal with http://ip-address:9001 from my mac, I get 
Bad Request
For request 'GET /' [Host not allowed: ip-address:9001]
If I do a local curl: 
curl -I http://localhost:9001
HTTP/1.1 303 See Other
Pragma: no-cache
Expires: 0
Location: /login?target=/
Set-Cookie: PLAY_SESSION=; Max-Age=0; Expires=Thu, 01 Jan 1970 00:00:00 GMT; Path=/
Set-Cookie: PLAY_FLASH=eyJhbGciOiJIUzI1NiJ9.eyJkYXRhIjp7ImFsZXJ0VHlwZSI6ImRhbmdlciIsImFsZXJ0TWVzc2FnZSI6IllvdSBhcmUgbm90IGxvZ2dlZCBpbi4gUGxlYXNlIGxvZ2luIHRvIGNvbnRpbnVlLiJ9LCJuYmYiOjE2ODY3NzAwMTUsImlhdCI6MTY4Njc3MDAxNX0.ik49ypRhSIy9H1OCKXl6LV4kRyXsTGnAbSzKCv5MOv0; SameSite=Lax; Path=/; HTTPOnly
Cache-Control: no-cache, no-store, must-revalidate
Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
X-Frame-Options: DENY
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
X-Permitted-Cross-Domain-Policies: master-only
Date: Wed, 14 Jun 2023 19:13:35 GMT
Content-Length: 0

There is no firewall running on the host.

I ran w3m on the server and was able to see the login page.
 w3m http://localhost:9001

Is there something else that needs to be configured to allow access?
In case it matters, the server is on one network and my mac is on another.  Interestingly, I am running nginx reverse proxy on the same docker and I can access the web page just fine.

Dave

**Comment by Aravindh-Raju** [comment]
Hi,

The error `'GET /' [Host not allowed: ip-address:9001]` usually means there's a problem with [Play's Allowed Host Filter](https://www.playframework.com/documentation/2.8.x/AllowedHostsFilter). 
I get the same error when I forward my localhost:9001 to a ngrok url as seen in the screenshot attached below:
<img width="1904" alt="Screenshot 2023-06-22 at 6 40 12 PM" src="https://github.com/vinyldns/vinyldns/assets/61419792/6f3d6e35-49da-429d-8688-74777d947a0f">

The solution is to add the [Play's Allowed Host Filter](https://www.playframework.com/documentation/2.8.x/AllowedHostsFilter) in your [application.conf](https://github.com/vinyldns/vinyldns/blob/master/modules/portal/conf/application.conf) present at `modules/portal/conf/`.
I added the following lines in my [application.conf](https://github.com/vinyldns/vinyldns/blob/master/modules/portal/conf/application.conf):
```
play.filters.hosts {
  allowed = ["."]
}
```
Note: `allowed = ["."]` is not recommended to use in production as it matches all hosts.

Here's the output of my ngrok portal after adding the above config:
<img width="1904" alt="Screenshot 2023-06-22 at 6 46 55 PM" src="https://github.com/vinyldns/vinyldns/assets/61419792/fcab443c-1345-48c1-88fa-f289672f7a57">

Additional reference: https://stackoverflow.com/questions/45070168/host-not-allowed-error-when-deploying-a-play-framework-application-to-amazon-a

Hope the solution works! Let us know if it works and if you still face any issues.

**Reply by mosespray** [reply]
Hello,
Thank you for the reply.  I tried what you suggested.  I added those lines to the config file, but had the same result.  Is there something that I need to do to make the changes take effect?  Other than just restarting the container?

D

**Reply by Aravindh-Raju** [reply]
Since you are using quickstart instructions and running the portal as a container too, can you please try adding the same config I mentioned above to application.conf in `build/docker/portal` (the file here: https://github.com/vinyldns/vinyldns/blob/master/build/docker/portal/application.conf) instead of `modules/portal/conf` that you tried now.

Once the change is done, remove the old images and run the quickstart command once again.

Let us know if this works.

**Comment by mosespray** [comment]
Hello again,

So, that worked for me.  Woot!  I have to wonder, though, how could anyone get this working without making that change?  Should that be in the quickstart instructions?  Just wondering.  Thank you very much for your help.  I really appreciate it.

D

**Reply by Aravindh-Raju** [reply]
Hi,

Good to know. Yes, it should be in the quickstart instructions or be added in application.conf file. We will make the necessary change soon. Thank you!
